File: joka/entities/migration.py
from dataclasses import dataclass

# statuses for applied migrations
STATUS_APPLIED = "applied"
STATUS_PENDING = "pending"
STATUS_OUT_OF_ORDER = "out_of_order"
STATUS_FILE_MISSING = "file_missing"

@dataclass
class Migration:
    """
    Representation of an applied migration.
    """
    id: int 
    migration_index: str
    applied_at: str  # ISO formatted datetime string
    file_name: str
    file_full_path: str
    status: str  # e.g., "applied", "pending"

File: internal/domains/migration/domain/migration.go
package domain

// Migration status constants used to classify each migration in the chain.
const (
	StatusApplied     = "applied"
	StatusPending     = "pending"
	StatusOutOfOrder  = "out_of_order"
	StatusFileMissing = "file_missing"
)

// Migration is the aggregate that combines database state and file state for
// a single migration, along with a computed status indicating whether it has
// been applied, is pending, or has a problem.
type Migration struct {
	ID             int
	MigrationIndex string
	AppliedAt      string // ISO formatted datetime string, empty if pending
	FileName       string
	FileFullPath   string
	Status         string // one of the Status* constants
}
